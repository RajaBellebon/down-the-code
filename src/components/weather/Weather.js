import React, { Component } from 'react';
import LatLong from './LatLong';
import DisplayTemp from './DisplayTemp';
const Data = require('./helper');


export default class Weather extends Component {
constructor(props){
    super(props);
    this.state ={
      Country: '',
      Lat: '',
      Long: '',
      Temp:'Loading',
      TempMin: '',
      TempMax: '',
      Sunrise: '',
      Sunset: '',
      Weather: {},
      Name: ''
    }
    this.onPass = this.onPass.bind(this)
  };
onPass(Lat,Long) {
    const that = this;
Data.getTemperature(Lat,Long).then(function (data) {
      that.setState({
        Lat,
        Long,
        Country: data.sys.country,
        Temp: data.main.temp,
        TempMin: data.main.temp_min,
        TempMax: data.main.temp_max,
        Sunrise: data.sys.sunrise,
        Sunset: data.sys.sunset,
        Weather: data.weather[0],
        Name: data.name
      });
    },
    function (errorMessage) {
        alert(errorMessage);
    });
  }
render(){
    return(
      <div>
        <LatLong onPass = {this.onPass} />
        <DisplayTemp Country={this.state.Country} Name = {this.state.Name} Lat = {this.state.Lat} Long = {this.state.Long} Temp = {this.state.Temp} TempMin = {this.state.TempMin} TempMax = {this.state.TempMax} Sunrise={this.state.Sunrise} Sunset={this.state.Sunset} Weather = {this.state.Weather}/>
      </div>
    );
  }
}
