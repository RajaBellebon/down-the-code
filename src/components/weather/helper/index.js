const axios = require('axios');

const OPEN_WEATHER_MAP_URL = `http://api.openweathermap.org/data/2.5/weather?&units=metric&APPID=${
  process.env.GATSBY_WEATHER_KEY
}`;

const getTemperature = (Lat, Long) => {
  const requestUrl = `${OPEN_WEATHER_MAP_URL}&lat=${Lat}&lon=${Long}`;
  return axios.get(requestUrl).then(
    function(res) {
      console.log(res.data);
      if (res.data.cod && res.data.message) {
        throw new Error(res.data.message);
      } else {
        console.log(res.data);
        return res.data;
      }
    },
    function(res) {
      throw new Error(res.data.message);
    },
  );
};

const feelingEmoji = temp => {
  let emoji;
  if (temp) {
    switch (true) {
      case temp >= 25:
        emoji = '🤩';
        break;
      case temp < 25 && temp >= 20:
        emoji = '😓';
        break;
      case temp < 20 && temp >= 0:
        emoji = '😰';
        break;
      case temp < 0:
        emoji = '🤧';
        break;
      default:
        emoji = '👋🏾';
    }
  } else {
    emoji = '😫';
  }
  return emoji;
};

const weatherEmoji = param => {
  let emoji;
  if (param) {
    const weatherDescription = param.toLowerCase();
    switch (true) {
      case weatherDescription.includes('rain'):
        emoji = '🌧 ☔️';
        break;
      case weatherDescription.includes('sun') &&
        (weatherDescription.includes('cloud') ||
          weatherDescription.includes('clouds')):
        emoji = '🌤⛅';
        break;
      case weatherDescription.includes('clouds') ||
        (weatherDescription.includes('cloud') &&
          !weatherDescription.includes('sun')):
        emoji = '☁️';
        break;
      case weatherDescription.includes('sun') &&
        !(
          weatherDescription.includes('cloud') ||
          weatherDescription.includes('clouds')
        ):
        emoji = '☀️';
        break;
      case weatherDescription.includes('clear'):
        emoji = '☀️';
        break;
      case weatherDescription.includes('snow'):
        emoji = '❄️';
        break;
      case weatherDescription.includes('fog'):
        emoji = '🌫';
        break;
      case weatherDescription.includes('wind') ||
        weatherDescription.includes('winds'):
        emoji = '🌬';
        break;
      default:
        emoji = '👋🏾';
    }
  } else {
    emoji = '😫';
  }

  return emoji;
};

module.exports = { getTemperature, feelingEmoji, weatherEmoji };
