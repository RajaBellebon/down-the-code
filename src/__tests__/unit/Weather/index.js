import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Weather from './../components/Weather/Weather';
import { feeling, weatherEmoji } from './../components/Weather/helper'

Describe('Weather Functions', () => {
  test('Feeling function should return default emoji 👋🏾)', () => {
    expect(feeling(undefined)).toBe('👋🏾');
  });
  test('Weather emoji function should return default emoji 👋🏾', () => {
    expect(weatherEmoji(undefined)).toBe('👋🏾');
  });
});


Describe('Weather Components', () => {
  const wrapper = mount(<Weather />);
  test('Weather component should render', () => {
    expect(wrapper).toBeDefined();
  });
  test('Weather component should match snapshots', () => {
    expect(wrapper).toMatchSnapshot();
  });
})