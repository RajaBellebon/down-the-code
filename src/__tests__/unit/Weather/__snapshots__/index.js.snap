// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Components Weather component should match snapshots when state changes 1`] = `
<Weather>
  <div>
    <LatLong
      onPass={[Function]}
    >
      <div />
    </LatLong>
    <DisplayTemp
      Country="AU"
      Lat={-33.9139319}
      Long={151.2423863}
      Name="Randwick"
      Sunrise=""
      Sunset=""
      Temp="Loading"
      TempMax=""
      TempMin=""
      Weather={Object {}}
    >
      <div
        className="loading"
      >
        Loading Your Local Weather...
      </div>
    </DisplayTemp>
  </div>
</Weather>
`;
